src/Nncase.Graph/Transform/RewriteProvider.cs

```diff
@@ -33,6 +34,7 @@ public Expr Rewrite(Expr expr, IEnumerable<IRewriteRule> rules, RunPassContext c
                 post = visitor.Rewrite(last);
                 if (visitor.IsMutated)
                 {
+                    currentRule = rule;
```

Comment:
Actual no need to use `currentRule`, when `IsMutated` the `lastRule` will be overridden by `rule`.